src/rgw/rgw_trim_mdlog.cc

```diff
@@ -677,9 +677,49 @@ class MetaPeerTrimPollCR : public MetaTrimPollCR {
   {}
 };
 
+namespace {
+bool sanity_check_endpoints(const DoutPrefixProvider *dpp, rgw::sal::RadosStore* store) {
+  bool retval = true;
+  auto current = store->svc()->mdlog->get_period_history()->get_current();
+  const auto& period = current.get_period();
+  for (const auto& [_, zonegroup] : period.get_map().zonegroups) {
+    if (zonegroup.endpoints.empty()) {
+      ldpp_dout(dpp, -1)
+	<< __PRETTY_FUNCTION__ << ":" << __LINE__
+	<< " ERROR: Cluster is is misconfigured! "
+	<< " Zonegroup " << zonegroup.get_name()
+	<< " ( " << zonegroup.get_id() << ") in Realm "
+	<< period.get_realm_name() << " ( " << period.get_realm() << ") "
+	<< " has no endpoints! Trimming is impossible." << dendl;
+      retval = false;
+    }
+    for (const auto& [_, zone] : zonegroup.zones) {
+      if (zone.endpoints.empty()) {
+	ldpp_dout(dpp, -1)
+	  << __PRETTY_FUNCTION__ << ":" << __LINE__
+	  << " ERROR: Cluster is is misconfigured! "
+	  << " Zone " << zone.name << " ( " << zone.id << ") in Zonegroup"
+	  << zonegroup.get_name() << " ( " << zonegroup.get_id()
+	  << ") in Realm " << period.get_realm_name()
+	  << " ( " << period.get_realm() << ") "
+	  << " has no endpoints! Trimming is impossible." << dendl;
+	retval = false;
+      }
+    }
+  }
+  return retval;
+}
+}
+
 RGWCoroutine* create_meta_log_trim_cr(const DoutPrefixProvider *dpp, rgw::sal::RadosStore* store, RGWHTTPManager *http,
                                       int num_shards, utime_t interval)
 {
+  if (sanity_check_endpoints(dpp, store)) {
+    ldpp_dout(dpp, -1)
+      << __PRETTY_FUNCTION__ << ":" << __LINE__
+      << " ERROR: Cluster is is misconfigured! Refusing to trim." << dendl;
+      return nullptr;
```

Comment:
`sanity_check_endpoints()` returns false when not sane, so this condition should be negated. same with the call below

Reply:
Well, that was embarrassing.